lms/djangoapps/grades/progress.py

```diff
@@ -0,0 +1,152 @@
+"""
+Progress Summary of a learner's course grades.
+"""
+from course_blocks.api import get_course_blocks
+from courseware.model_data import ScoresClient
+from student.models import anonymous_id_for_user
+from util.db import outer_atomic
+from xmodule import graders, block_metadata_utils
+from xmodule.graders import Score
+
+from .scores import get_score, possibly_scored
+
+
+class ProgressSummary(object):
+    """
+    Wrapper class for the computation of a user's scores across a course.
+
+    Attributes
+       chapters: a summary of all sections with problems in the course. It is
+       organized as an array of chapters, each containing an array of sections,
+       each containing an array of scores. This contains information for graded
+       and ungraded problems, and is good for displaying a course summary with
+       due dates, etc.
+
+       weighted_scores: a dictionary mapping module locations to weighted Score
+       objects.
+
+       locations_to_children: a function mapping locations to their
+       direct descendants.
+    """
+    def __init__(self, chapters=None, weighted_scores=None, locations_to_children=None):
+        self.chapters = chapters
+        self.weighted_scores = weighted_scores
+        self.locations_to_children = locations_to_children
+
+    def score_for_module(self, location):
+        """
+        Calculate the aggregate weighted score for any location in the course.
+        This method returns a tuple containing (earned_score, possible_score).
+
+        If the location is of 'problem' type, this method will return the
+        possible and earned scores for that problem. If the location refers to a
+        composite module (a vertical or section ) the scores will be the sums of
+        all scored problems that are children of the chosen location.
+        """
+        if location in self.weighted_scores:
+            score = self.weighted_scores[location]
+            return score.earned, score.possible
+        children = self.locations_to_children[location]
+        earned = 0.0
+        possible = 0.0
+        for child in children:
+            child_earned, child_possible = self.score_for_module(child)
+            earned += child_earned
+            possible += child_possible
+        return earned, possible
+
+
+def summary(student, course, course_structure=None):
```

Comment:
Regarding the name `summary`, I actually thought having the same "namespaced" name for both made sense rather than trying to concoct 2 different names just to be different.  And they are essentially both creating a summary dictionary of the grades data across the course.

So it is currently: `progress.summary` and `course_grades.summary`.

But I'm definitely open to further discussion if you have other thoughts.